src/libxtp/gwbse/sigma_exact.cc

```diff
@@ -97,13 +97,12 @@ Eigen::MatrixXd Sigma_Exact::CalcResidues(Index gw_level) const {
   const Index qpoffset = _opt.qpmin - _opt.rpamin;
   const Index auxsize = _Mmn.auxsize();
   vc2index vc = vc2index(0, 0, n_unocc);
-  const Eigen::MatrixXd Mmn_i_T = _Mmn[gw_level + qpoffset].transpose();
+  const Eigen::MatrixXd& Mmn_i = _Mmn[gw_level + qpoffset];
   Eigen::MatrixXd res = Eigen::MatrixXd::Zero(_rpatotal, rpasize);
   for (Index v = 0; v < n_occ; v++) {  // Sum over v
-    const Eigen::MatrixXd& Mmn_v = _Mmn[v].block(n_occ, 0, n_unocc, auxsize);
-    const Eigen::MatrixXd fc = Mmn_v * Mmn_i_T;  // Sum over chi
-    const Eigen::MatrixXd& XpY_v =
-        _rpa_solution.XpY.block(vc.I(v, 0), 0, n_unocc, rpasize);
+    auto Mmn_v = _Mmn[v].block(n_occ, 0, n_unocc, auxsize);
```

Comment:
Why use auto keyword? Does it prevent Eigen from casting their result into a matrix?

Reply:
yeah it is just an expression, I do not want a copy.